pkgs/development/python-modules/speechrecognition/default.nix

```diff
@@ -39,26 +44,31 @@ buildPythonPackage rec {
   build-system = [ setuptools ];
 
   dependencies = [
+    pyaudio
     requests
     typing-extensions
   ];
 
-  nativeCheckInputs = [
-    numpy
-    pytestCheckHook
-    torch
-    soundfile
-  ];
+  optional-dependencies = {
+    amazone = [ boto3 ];
+    google-cloud-speech = [ google-cloud-speech ];
+    lex = [ boto3 ];
+    pocketsphinx = [ pocketsphinx ];
+    tensorflow = [ tensorflow ];
+    whisper-api = [ openai ];
+    whisper-local = [
+      openai-whisper
+      soundfile
+    ];
+  };
+
+  nativeCheckInputs = [ pytestCheckHook ] ++ lib.flatten (builtins.attrValues optional-dependencies);
```

Comment:
I think we just want to list the optional dependencies for tests here only

Reply:
Done, reverted to how I had it before.